src/main/java/io/github/jsonSnapshot/SnapshotDataItem.java

```diff
@@ -0,0 +1,67 @@
+package io.github.jsonSnapshot;
+
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import lombok.NonNull;
+import lombok.Value;
+
+@Value
+public class SnapshotDataItem implements Comparable<SnapshotDataItem> {
+  private static final int REGEX_FLAGS = Pattern.MULTILINE + Pattern.DOTALL;
+  private static final Pattern REGEX =
+      Pattern.compile("(?<name>[^ =]*) *=+ *(?<data>\\[.*\\])[^\\]]*", REGEX_FLAGS);
+
+  public static SnapshotDataItem ofRawData(@NonNull final String rawDataString)
+      throws IllegalRawDataException {
+    final Matcher matcher = REGEX.matcher(rawDataString);
+    if (!matcher.matches()) {
+      throw new IllegalRawDataException(rawDataString);
+    }
+
+    final String name = matcher.group("name");
+    final String data = matcher.group("data");
+    return SnapshotDataItem.ofNameAndData(trimOrNull(name), trimOrNull(data));
+  }
+
+  private static String trimOrNull(final String str) {
+    if (str == null) {
+      return null;
+    }
+    return str.trim();
+  }
+
+  public static SnapshotDataItem ofNameAndData(
+      @NonNull final String name, @NonNull final String data) {
+    return new SnapshotDataItem(name, data);
+  }
+
+  String name;
```

Comment:
I'd make the fields `private` and `final` for those not familiar with Lombok.
Usually, fields in a class are placed before any methods (even static ones)

Reply:
done (both)